lua/neo-tree/ui/renderer.lua

```diff
@@ -1010,6 +1013,9 @@ M.acquire_window = function(state)
       vim.api.nvim_win_set_buf(state.winid, state.bufnr)
     else
       close_old_window()
+      if state.bufnr and vim.api.nvim_buf_is_valid(state.bufnr) then
```

Comment:
`M.close` is already deleting the buffer. Why are we double deleting? Is it because of your other change which will only sometimes delete the old buffer? It seems like these two changes are in conflict with another.

Also, if i remember correctly, the original code from a few months ago already did what we are slowly getting back to, which is to immediately attempt to delete the buffer and then schedule a delete for later only if we get a specific error.

I think the right place to make this fix is in the `M.close()` code, we shouldn't be deleting the buffer multiple times and from different locations.